client/rest-high-level/src/main/java/org/elasticsearch/client/asyncsearch/SubmitAsyncSearchRequest.java

```diff
@@ -0,0 +1,282 @@
+    }
+
+    /**
+     * Specifies what type of requested indices to ignore and how to deal with indices wildcard expressions.
+     */
+    public void setIndicesOptions(IndicesOptions indicesOptions) {
+        this.searchRequest.indicesOptions(indicesOptions);
+    }
+
+    /**
+     * Get the indices Options.
+     */
+    public IndicesOptions getIndicesOptions() {
+        return this.searchRequest.indicesOptions();
+    }
+
+    /**
+     * The search type to execute, defaults to {@link SearchType#DEFAULT}.
+     */
+    public void setSearchType(SearchType searchType) {
+        this.searchRequest.searchType(searchType);
+    }
+
+    /**
+     * Get the search type to execute, defaults to {@link SearchType#DEFAULT}.
+     */
+    public SearchType getSearchType() {
+        return this.searchRequest.searchType();
+    }
+
+    /**
+     * Sets if this request should allow partial results. (If method is not called,
+     * will default to the cluster level setting).
+     */
+    public void setAllowPartialSearchResults(boolean allowPartialSearchResults) {
+        this.searchRequest.allowPartialSearchResults(allowPartialSearchResults);
+    }
+
+    /**
+     * Gets if this request should allow partial results.
+     */
+    public Boolean getAllowPartialSearchResults() {
+        return this.searchRequest.allowPartialSearchResults();
+    }
+
+    /**
+     * Sets the number of shard results that should be reduced at once on the coordinating node. This value should be used as a protection
+     * mechanism to reduce the memory overhead per search request if the potential number of shards in the request can be large.
+     */
+    public void setBatchedReduceSize(int batchedReduceSize) {
+        this.searchRequest.setBatchedReduceSize(batchedReduceSize);
+    }
+
+    /**
+     * Gets the number of shard results that should be reduced at once on the coordinating node.
+     * This defaults to 5 for {@link SubmitAsyncSearchRequest}.
+     */
+    public int getBatchedReduceSize() {
+        return this.searchRequest.getBatchedReduceSize();
+    }
+
+    /**
+     * Sets if this request should use the request cache or not, assuming that it can (for
+     * example, if "now" is used, it will never be cached). By default (not set, or null,
+     * will default to the index level setting if request cache is enabled or not).
+     */
+    public void setRequestCache(Boolean requestCache) {
+        this.searchRequest.requestCache(requestCache);
+    }
+
+    /**
+     * Gets if this request should use the request cache or not.
+     * Defaults to `true` for {@link SubmitAsyncSearchRequest}.
+     */
+    public Boolean getRequestCache() {
+        return this.searchRequest.requestCache();
+    }
+
+    /**
+     * Gets if the source of the {@link SearchSourceBuilder} initially used on this request.
+     */
+    public SearchSourceBuilder getSearchSource() {
+        return this.searchRequest.source();
+    }
```

Comment:
we may want to expose a setter for the source too. Not so important but I guess users would argue that once set at the constructor it can never be reset. Search request allows to set it.

Reply:
Sorry, I don't follow. Why would you want to change the source when you are constructing the request in the client? You can get it and change the various parts of it if you like. Why aim for parity with SearchRequest? This would mean allowing the user to reach down into a thing we want to hide (the SearchRequest) and make changes there. I don't see how thats useful.

Reply:
I would have replied the same way to this comment :) it is just a comment that I got in the past with these client requests. People expect to be able to reset stuff, while here once you set the source you can not go back to the initial state. Though you can modify it. I am fine with doing without the setter though, it is not that useful.